docs/how-to-guides/feathr-udfs.md

```diff
@@ -0,0 +1,182 @@
+  WHERE DOLocationID!= 100
+  """
+  )
+  return sqlDF
+```
+
+### UDF with Pandas Example:
+
+Feathr also supports using pandas to deal with the data. Behind the scene it's using pandas-on-spark so some limitation applies here. Please refer to [Pandas-on-Spark's Best Practice](https://spark.apache.org/docs/latest/api/python/user_guide/pandas_on_spark/best_practices.html#best-practices) for more details.
+
+Below is the template, note that:
+
+1. This is only available for Spark 3.2 and later, so make sure you submit to a spark cluster that has Spark 3.2 and later.
+2. You need to call something like `psdf = df.to_pandas_on_spark()` to convert a Spark dataframe to "pandas dataframe", and call `psdf.to_spark()` to convert the "pandas dataframe" to Spark dataframe.
+
+```python
+def feathr_udf_pandas_spark(df: DataFrame) -> DataFrame:
+  # using pandas on spark APIs. Fore more details, refer to the doc here: https://spark.apache.org/docs/latest/api/python/user_guide/pandas_on_spark/index.html
+  # Note that this API is only available for Spark 3.2 and later, so make sure you submit to a spark cluster that has Spark 3.2 and later.
+  psdf = df.to_pandas_on_spark()
+  psdf['fare_amount_cents'] = psdf['fare_amount']*100
+  # need to make sure converting the pandas-on-spark dataframe to Spark Dataframe.
+  return psdf.to_spark()
+
+batch_source = HdfsSource(name="nycTaxiBatchSource",
+                        path="abfss://feathrazuretest3fs@feathrazuretest3storage.dfs.core.windows.net/demo_data/green_tripdata_2020-04.csv",
+                        preprocessing=add_new_dropoff_and_fare_amount_column,
+                        event_timestamp_column="new_lpep_dropoff_datetime",
+                        timestamp_format="yyyy-MM-dd HH:mm:ss")
+```
+
+### When do I use UDFs?
+
+Those UDFs are totally optional to use. For example, if you have an existing feature transformation pipeline, you don't have to use Feathr's preprocessing functions to rewrite your code. Instead, you can simply use your already transformed feature in Feathr, for point in time joins, or for feature registry and exploration.
+
+But if you don't have an existing pipeline, Feathr's UDF does provide a good way for you to manage your feature engineering system from end to end. This decision is beyond the scope of this document.
+
+## UDFs at individual features (using the `transform` parameters)
+
+Other than the UDF in the input level, Feathr also allows some level of customizations in the individual feature level. This is very useful if users want to do some last mile transformation, or for derived features where users might want to compose a new feature based on a few existing features with some calculation.
+
+For those row-level transformations, [Spark SQL built-in functions](https://spark.apache.org/docs/latest/api/sql/index.html) are supported. For example you can call `dayofmonth`, `xpath_double`, or `percent_rank` etc. in the Spark SQL built-in functions.
```

Comment:
>For example you can call `dayofmonth`, `xpath_double`, or `percent_rank` etc. in the Spark SQL built-in functions. [](http://example.com/codeflow?start=140&length=114)

Do you meancan call Spark SQL built-in functions dayofmonth, xpath_double, or percent_rank etc  in transofrm expression?